### ISSUE TITLE
First attempt to build setup the requirements in linux

### ISSUE THREAD

**Issue by tiborkiss** [TheSoundOfAIOSR/rg_production#18]
I used venv and pip with python 3.8.

Do you know for these unsuccessful build there are some description, which I have to look and manually pre-install something?
csound is natively installed with apt.

Here is the output where the requirements setup fails.
```
Building wheels for collected packages: Kivy-Garden, ctcsound, python-rtmidi, librosa, audioread, resampy
  Building wheel for Kivy-Garden (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/tibor/venv/rgprod/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5joa18yh/Kivy-Garden/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5joa18yh/Kivy-Garden/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-dqp0g1fx
       cwd: /tmp/pip-install-5joa18yh/Kivy-Garden/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help

  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for Kivy-Garden
  Running setup.py clean for Kivy-Garden
  Building wheel for ctcsound (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/tibor/venv/rgprod/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5joa18yh/ctcsound/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5joa18yh/ctcsound/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-4hj2wtte
       cwd: /tmp/pip-install-5joa18yh/ctcsound/
  Complete output (7 lines):
  Warning: 'classifiers' should be a list, got type 'filter'
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help

  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for ctcsound
  Running setup.py clean for ctcsound
  Building wheel for python-rtmidi (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/tibor/venv/rgprod/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5joa18yh/python-rtmidi/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5joa18yh/python-rtmidi/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-qzh29qqv
       cwd: /tmp/pip-install-5joa18yh/python-rtmidi/
  Complete output (8 lines):
  Detected JACK version 1.9.12.
  JACK version is recent enough to have 'jack_port_rename' function.
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help

  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for python-rtmidi
  Running setup.py clean for python-rtmidi
  Building wheel for librosa (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/tibor/venv/rgprod/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5joa18yh/librosa/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5joa18yh/librosa/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-qc16ay6q
       cwd: /tmp/pip-install-5joa18yh/librosa/
  Complete output (6 lines):
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help

  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for librosa
  Running setup.py clean for librosa
  Building wheel for audioread (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/tibor/venv/rgprod/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5joa18yh/audioread/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5joa18yh/audioread/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-je4oqaki
       cwd: /tmp/pip-install-5joa18yh/audioread/
  Complete output (8 lines):
  /tmp/pip-install-5joa18yh/audioread/setup.py:17: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help

  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for audioread
  Running setup.py clean for audioread
  Building wheel for resampy (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/tibor/venv/rgprod/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-5joa18yh/resampy/setup.py'"'"'; __file__='"'"'/tmp/pip-install-5joa18yh/resampy/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-dkonawhl
       cwd: /tmp/pip-install-5joa18yh/resampy/
  Complete output (8 lines):
  /tmp/pip-install-5joa18yh/resampy/setup.py:2: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp
  usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: setup.py --help [cmd1 cmd2 ...]
     or: setup.py --help-commands
     or: setup.py cmd --help

  error: invalid command 'bdist_wheel'
  ----------------------------------------
  ERROR: Failed building wheel for resampy
  Running setup.py clean for resampy
Failed to build Kivy-Garden ctcsound python-rtmidi librosa audioread resampy
Installing collected packages: pygments, certifi, urllib3, idna, chardet, requests, Kivy-Garden, docutils, kivy, pycparser, CFFI, sounddevice, numpy, ctcsound, python-rtmidi, audioread, decorator, joblib, llvmlite, numba, appdirs, pyparsing, packaging, pooch, scipy, six, resampy, threadpoolctl, scikit-learn, soundfile, librosa
    Running setup.py install for Kivy-Garden ... done
    Running setup.py install for ctcsound ... done
    Running setup.py install for python-rtmidi ... done
    Running setup.py install for audioread ... done
    Running setup.py install for resampy ... done
    Running setup.py install for librosa ... done
Successfully installed CFFI-1.14.5 Kivy-Garden-0.1.4 appdirs-1.4.4 audioread-2.1.9 certifi-2020.12.5 chardet-4.0.0 ctcsound-6.14.2 decorator-4.4.2 docutils-0.16 idna-2.10 joblib-1.0.1 kivy-2.0.0 librosa-0.8.0 llvmlite-0.36.0 numba-0.53.0 numpy-1.20.1 packaging-20.9 pooch-1.3.0 pycparser-2.20 pygments-2.8.1 pyparsing-2.4.7 python-rtmidi-1.4.7 requests-2.25.1 resampy-0.2.2 scikit-learn-0.24.1 scipy-1.6.2 six-1.15.0 sounddevice-0.4.1 soundfile-0.10.3.post1 threadpoolctl-2.1.0 urllib3-1.26.4
```

**Comment by Redict** [comment]
Can you tell me more about how you caused the errors?

**Comment by tiborkiss** [comment]
I figured out how to setup correctly.

Before creating the venv, I had to install the following
```
sudo apt-get install csound
sudo apt-get install libasound2-dev
sudo apt-get install libjack-jackd2-dev
```
I used Python 3.8 with the built-in `venv`.  After I create the venv and activate, I must upgrade pip
```
pip install --upgrade pip
```
Then `pip install -r requirements.txt` will not fail with the setup.

Overall, I was able to install on 
 - native Ubuntu 20.04 LTS
 - Ubuntu 20.04 LTS running inside WSL2 (Insiders version, based on Build 21322.rs_prerelease Windows 10)

With that above, I see no reason to not work in any of the systems we use. Therefore I close the issue also.
